core/trino-main/src/main/java/io/trino/execution/SqlTask.java

```diff
@@ -193,6 +193,12 @@ private void initialize(Function<SqlTask, ?> onDone, CounterStat failedTasks)
         });
     }
 
+    @SuppressWarnings("ReturnValueIgnored") // onDone should probably be a Consumer, but that would change the public API
```

Comment:
What public API do you have in mind?

Reply:
I'm being very cautious here, I'm trying to avoid modifying any of the `public` mehods, in case this is not just an internal API. Yes, this is confusing in Java and modules were supposed to make it easier to declare what is and what is not a user-visible "public API" that is not supposed to be changed.

Reply:
In this case this is used only by the `SqlTaskManager` and two tests, so I made the method package-private and replaced `Function` with a `Consumer`. If this is in fact modifying a public API I expect to hear about this ;)

Also, moved to #6898

Reply:
In a single module library-like project, every public method can be perhaps considered an API.
In Trino case, things are obviously more complex, but at the end of the day -- not very complex.
In particular, internal classes in trino-main, that are not exposed outside the server, are not part of the API